frontend/src/components/nodes/Validators.tsx

```diff
@@ -1,27 +1,116 @@
 import React from "react";
 
-import * as N from "../../libraries/explorer-wamp/nodes";
 import { NodeConsumer } from "../../context/NodeProvider";
 
-import ValidatorRow from "./ValidatorRow";
+import { onPageChange as P } from "../utils/Pagination";
+import { Table } from "../utils/Table";
 import PaginationSpinner from "../utils/PaginationSpinner";
 
-class Validators extends React.PureComponent {
+import ValidatorsList from "./ValidatorsList";
+
+interface Props {
+  type: string;
+  itemsPerPage: number;
+}
+
+class Validators extends React.PureComponent<Props> {
+  static defaultProps = {
+    itemsPerPage: 12,
+  };
+
+  state = {
+    activePage: 0,
+    startPage: 1,
+    endPage: this.props.itemsPerPage,
+  };
+  onPageChange = ({ selected }: P) => {
+    const { itemsPerPage } = this.props;
+    this.setState({
+      activePage: selected,
+      startPage: selected * itemsPerPage + 1,
+      endPage: selected * itemsPerPage + itemsPerPage,
+    });
+  };
+
   render() {
+    const { activePage, startPage, endPage } = this.state;
+    const { type, itemsPerPage } = this.props;
+
+    const tableColumns =
+      type === "validators"
+        ? [
+            " ",
+            "#",
+            "Validator",
+            "Fee",
+            "Delegators",
+            "Stake",
+            "Cumulative Stake",
+          ]
+        : [" ", "#", "Validator", "Fee", "Delegators", "Stake"];
```

Comment:
I would avoid all this complexity and "generic" thing, and render the columns directly using `th` below instead of crafting this array first, and then iterate over it, and they apply className conditionally. It seems to be more complex than it is necessary